### ISSUE TITLE
theme-preview.sh reports ./liquidprompt: line 2461: lp_temperature: unbound variable

### ISSUE THREAD

**Issue by LudovicRousseau** [nojhan/liquidprompt#670]
```
$ LANG=C ./tools/theme-preview.sh default
Short prompt:

  [user:~] $   

Medium prompt:

  1& [user@server:~/liquidprompt] main ±   

./liquidprompt: line 2461: lp_temperature: unbound variable
```

Shell: 
```
± echo $BASH_VERSION
5.0.3(1)-release
```

Operating system: 
```
± lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
```

Liquidprompt version: 
± git rev-parse HEAD
a5ecd88e52d83de37cb6b4fb598c2e53748d1f2f
± git describe --tags --exact-match
v2.0.2

### Steps to Reproduce

1. run `./tools/theme-preview.sh default`

I have 2 systems running Debian stable so the same version of Bash. Also the same version of liquidprompt.
I have the problem only on 1 of the 2 systems.

### Expected Behavior
No error

### Current Behavior
`./liquidprompt: line 2461: lp_temperature: unbound variable`

### Possible Solution
The problem comes from `set -u` in `./tools/theme-preview.sh`.

I reported a similar problem in https://github.com/nojhan/liquidprompt/issues/656

**Comment by Rycieos** [comment]
I can't reproduce on any version of Bash on commit a5ecd88e52d83de37cb6b4fb598c2e53748d1f2f or 964a4600f79bd84c71218feadb3c37f72e37fc27.

Could you paste a debug output?
```bash
LANG=C bash -x ./tools/theme-preview.sh default
```

**Comment by Rycieos** [comment]
Ah, that makes perfect sense. `theme-preview.sh` overrides `_lp_temperature()`, but I had forgotten about the activate checking which temp backends were valid.

 `__lp_temp_detect()` then throws that error, because `lp_temperature` was never set, because the only value it got was 0, which failed the compare. This bug is also in normal Liquidprompt, but since you don't run your shell with `set -u` you aren't seeing it.

This will be simple to fix. I'm assuming you would like a fix backported to the v2.0 line?

**Comment by LudovicRousseau** [comment]
I don't have any issue with v2.0.2 in my normal shell. So I do not need a backported patch.
I will upgrade to a fixed version when it is ready.

**Comment by Rycieos** [comment]
Fixed in 4c81ef4026c3105a260ff2b7e275bc8e50921c3b. Added a lot of test coverage to validate and prevent regression.

**Comment by Rycieos** [comment]
I cherry-picked this back for [v2.0.3](https://github.com/nojhan/liquidprompt/releases/tag/v2.0.3), since there was another, more critical issue that forced a patch on v2.0.
